src/utils/hooks/useLoadApplicationNEW.ts

```diff
@@ -13,34 +13,60 @@ import { buildSectionsStructure } from '../helpers/structure/buildSectionsStruct
 import {
   Application,
   ApplicationSection,
+  ApplicationStageStatusAll,
+  ApplicationStatus,
   Template,
   TemplateElement,
   TemplateStage,
   useGetApplicationNewQuery,
 } from '../generated/graphql'
+import messages from '../messages'
+
+const MAX_REFETCH = 10
 
 const useLoadApplication = ({ serialNumber, networkFetch }: UseGetApplicationProps) => {
   const [isLoading, setIsLoading] = useState(true)
   const [structureError, setStructureError] = useState('')
   const [structure, setFullStructure] = useState<FullStructure>()
   const [template, setTemplate] = useState<TemplateDetails>()
+  const [refetchAttempts, setRefetchAttempts] = useState(0)
 
-  const { data, error } = useGetApplicationNewQuery({
+  const { data, loading, error, refetch } = useGetApplicationNewQuery({
     variables: {
       serial: serialNumber,
     },
     fetchPolicy: networkFetch ? 'network-only' : 'cache-first',
+    notifyOnNetworkStatusChange: true,
   })
 
   useEffect(() => {
-    if (!data) return
+    if (!data || loading) return
     const application = data.applicationBySerial as Application
 
     if (!application) {
       setIsLoading(false)
       return
     }
 
+    if (!application.template) {
+      setStructureError(messages.APPLICATION_MISSING_TEMPLATE)
+      return
+    }
+
+    // Checking if trigger is running before loading current status
+    if (application.trigger === null) {
+      setRefetchAttempts(0)
+    } else {
+      if (refetchAttempts < MAX_REFETCH) {
+        setTimeout(() => {
+          console.log('Will refetch loadApplication', refetchAttempts) // TODO: Remove log
+          setRefetchAttempts(refetchAttempts + 1)
+          refetch()
+        }, 500)
```

Comment:
one more change please, setIsLoading = true somewhere here (to indicate that we are awaiting trigger even though loading = false (i.e. when submission button was pressed)

Reply:
or maybe this should be just before  ` if (!application.template) `{ ?

And I'v just noticed, 

```
f (!application) {
      setIsLoading(false)
      return
    }
```

Should have a setStructureError ?

Reply:
It is already set to true when the hook starts (isLoading)

Reply:
I am think of a use case when hook is re-run due to changes to application (when submitting)

Reply:
I see! I was trying to understand the use case. I'll add as part of the statement that checks trigger - if it needs to re-run will be set to true. Does that sound good?

Reply:
Done: 2c4c3fe